integration/test/sameclusterid/main_test.go

```diff
@@ -0,0 +1,30 @@
+// +build k8srequired
+
+package sameclusterid
```

Comment:
I don't like this name. Suggestions welcome.

Reply:
Is this a test we always have to have or will we be able to get rid again of it eventually?

Reply:
We had a chat. This test is going to remain. It will prevent us from such a regression in the future.

Reply:
https://github.com/giantswarm/aws-operator/pull/1214#discussion_r228103552

I think `recreatecluster` is better name.